### PR TITLE
Draft: Braintree android updates

### PR DESCRIPTION
**Pull request by thanh-fluz**

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -21,10 +21,11 @@ android {
...
     implementation 'com.google.android.gms:play-services-wallet:18.0.0'
-    implementation 'com.braintreepayments.api:google-payment:3.3.1'
-    implementation 'com.braintreepayments.api:data-collector:3.21.1'
+    // implementation 'com.braintreepayments.api:google-payment:4.45.0'
+    implementation 'com.braintreepayments.api:google-pay:4.45.0'
+    implementation 'com.braintreepayments.api:data-collector:4.45.0'
     implementation "com.facebook.react:react-native:+"
-    implementation 'com.braintreepayments.api:braintree:3.21.1'
-    implementation 'com.braintreepayments.api:three-d-secure:3.21.1'
+    // implementation 'com.braintreepayments.api:braintree:4.45.0'
```
Comment (fluz-jeff) [inline android/build.gradle]: `rm` commented code here and above

**Review by fluz-jeff — APPROVED** [review]

File: android/build.gradle (hunk 1 above)
Comment (thanh-fluz) [reply android/build.gradle]: yup, ill get it removed after android testing. right now still not 100% sure if it is not needed

**Comment by thanh-fluz** [comment]
We should also update the readme as much as possible too

File: android/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
         android:theme="@android:style/Theme.Translucent.NoTitleBar"
         android:exported="true" >
         
         <intent-filter>
             <action android:name="android.intent.action.VIEW" />
             <category android:name="android.intent.category.DEFAULT" />
-            <category android:name="android.intent.category.BROWSABLE" />
-            <data android:scheme="${applicationId}.braintree" />
-        </intent-filter>
+            <category android:applicationId}.braintree" />
```
Comment (thanh-fluz) [inline android/src/main/AndroidManifest.xml]: Need to revert

**Comment by thanh-fluz** [comment]
We may actually catch up the the braintree ios version v6.18.0 by the time testing is finished

image

File: android/src/main/java/com/pw/droplet/braintree/Braintree.java [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 import androidx.fragment.app.FragmentActivity;
 
-import com.braintreepayments.api.BraintreeFragment;
+import com.braintreepayments.api.Client;
```
Comment (thanh-fluz) [inline android/src/main/java/com/pw/droplet/braintree/Braintree.java]: decoupled BraintreeFragment incorrectly implemented. Each service needs to have its own Client. eg. GooglePayClient. Need to make change

**Comment by thanh-fluz** [comment]
@fluz-jeff @StronkMan Made a lot of android changes. view last 5 commits

File: android/src/main/java/com/pw/droplet/braintree/Braintree.java [hunk 4]
```diff
@@ -1,301 +1,354 @@
...
+
+        nonceErrorCallback(gson.toJson(errors));
+      } else {
+        nonceErrorCallback(errorWithResponse.getErrorResponse());
+      }
     }
+  }
+
+  @ReactMethod
+  public void setup(final String token, final Callback successCallback,
```
Comment (StronkMan) [inline android/src/main/java/com/pw/droplet/braintree/Braintree.java]: Can we move this method back to the top, before ln108.

Comment (thanh-fluz) [reply android/src/main/java/com/pw/droplet/braintree/Braintree.java]: done

**Review by StronkMan — APPROVED** [review]

**Pull request closed without merging**